velox/functions/prestosql/Utf8Utils.cpp

```diff
@@ -100,6 +141,35 @@ int32_t tryGetCharLength(const char* input, int64_t size) {
     return -4;
   }
 
-  VELOX_UNREACHABLE();
+  if (size < 5) {
+    return -4;
+  }
+
+  auto filefthByte = input[4];
+  if (!utf_cont(filefthByte)) {
+    return -4;
+  }
+
+  if (charLength == 5) {
+    // Per RFC3629, UTF-8 is limited to 4 bytes, so more bytes are illegal.
```

Comment:
if anything above 4 is illegal do we need any branches after 4? can we just return -1 after size or charLength > 4 ?
if not, do we need a utf_cont for those extra 5th and 6th bytes?

Reply:
Byte sequences of length 5 and 6 were legal in the past, but became illegal since RFC3629. However, the data we read can still have byte sequences of length 5 and 6 and we need to consider the sequence as one codepoint. This happens in the Presto batch query and Presto-java consider a sequence of 5 bytes as one codepoint .

Reply:
Thanks, can you clarify which a short comment as to what the output of this function signifies? I wasent entirely sure why we needed multiple negative values.

Reply:
This works, thanks for letting me know, i had not checked the header file earlier.